### PR TITLE
Added MSSQL Server for tests

### PR DESCRIPTION
**Pull request by jabolina**
Added MSSQL container support. Disabled by default, requires ISPN fix to work, and requires a cache-manager image.

### CODE REVIEW

**Comment by jabolina** [comment]
Just realized we have a cache-manager with MSSQL. Updating.

**Comment by jabolina** [comment]
Had an issue with the way Debezium exposes SQL Server metrics. Unfortunately, we are missing a few streaming metrics (looks like the ones which are static). By registering including the database name, we are retrieving the most important ones. I created #96 to fix that.

File: src/test/resources/mssql/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+if ! [ -f /var/opt/mssql/.initialized ]; then
+  while ! </dev/tcp/localhost/1433 2>/dev/null; do
+    sleep 2
+  done
+  # Just waiting a little to be sure.
+  sleep 5
```
Comment (jabolina) [inline src/test/resources/mssql/entrypoint.sh:8]: I am not liking this, we'll have flaky tests.

Comment (ryanemerson) [reply src/test/resources/mssql/entrypoint.sh:8]: I don't think there's any need to have a custom entry point for the DB. Instead we can create the testcontainer normally and then call `org.testcontainers.ext.ScriptUtils.runInitScript(...)` after calling `JdbcDatabaseContainer#start` in  order to execute `.setup.sql`.

We take this approach in the cache-manager:

https://github.com/gingersnap-project/cache-manager/blob/354fd06c2b2a7f2b3a4c0642d30fe5d8490b0a0a/manager/src/test/java/io/gingersnapproject/mssql/MSSQLServerResources.java#L17-L29

Comment (jabolina) [reply src/test/resources/mssql/entrypoint.sh]: Thanks a lot!! I updated to use the `withInitScript`, let's see how it goes with CI.

**Comment by jabolina** [comment]
@ryanemerson Seems to be passing. I am afraid the tests I added for the CDC events might be flaky, but I hope it doesn't take more than 15s between a change in the DB and publishing the event to the cache-manager.

**Comment by ryanemerson** [comment]
Better to have some flaky test coverage opposed to non at all. We can refine this as we go. Thanks @jabolina

**Pull request merged**